### DISCUSSION TITLE
Edit Functions

### DISCUSSION THREAD

**Discussion by mrb0nj** [supabase/supabase#7470] [Feature Requests]
It would be great if there was an option to edit a function directly in the sql editor rather than the flyout/drawer... 

with my limited experience so far, I've run into 2 situations where this would be useful

1) When i've either not set correctly/or need to change some of the advanced settings - security invoker/definer is the biggest offender lol
2) When the function gets a bit complex to view in the smaller edit view

**Comment by GaryAustin1** [comment]
You can create, alter or delete functions in the sql editor. 
“create or replace” is commonly used for this.

**Reply by mrb0nj** [reply]
Sure, this is how i've been doing it this way when i need to - I just thought it would be a nice shortcut to have that "create or replace" script populated in the editor at the click of a button.

**Comment by yanapaqui** [comment]
Any way of doing this????
Is really annoying to use other software to copy sql code function and paste it in the sql editor

**Reply by GaryAustin1** [reply]
No changes as of last time I used the Function editor a month ago.

**Comment by k2xl** [comment]
The issue I have is that I can't modify the function parameters (at least as far as I can tell) and their types. Often times i generate from an AI tool and then i find out that it doesn't use the right schema type.

**Reply by GaryAustin1** [reply]
You have to use the SQL editor to replace the function if you change the parameters.

**Reply by k2xl** [reply]
> You have to use the SQL editor to replace the function if you change the parameters.

Just curious why that is? Sometimes I forget the parameters - and it is a bit cumbersome to have to lookup how to get the existing function parameters each time (especially when using an LLM)

**Reply by GaryAustin1** [reply]
I don't know why Supabase decides to implement or not certain features.  The UI gets better all the time so I expect they have some priority on where to spend their resources versus benefit to the most users.
